l10n_it_delivery_note/models/stock_delivery_note.py

```diff
@@ -157,6 +157,7 @@ def _domain_weight_uom(self):
         required=True,
         index=True,
         check_company=True,
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
```

Comment:
Ho verificato in runboat e i tipi vengono già filtrati correttamente:
https://github.com/OCA/l10n-italy/assets/143788994/2ef1855b-7e74-4446-9639-87533afb4e9d
```suggestion
```

Il dominio viene aggiunto perché è impostato `check_company=True`, vedi in https://github.com/odoo/odoo/blob/04d5b0d863634652d14eeb822eabd27627567e1f/odoo/fields.py#L2540.

Aggiungendo `_check_company_auto = True` nel model, il dominio viene applicato anche durante la creazione dei record lato codice.

Reply:
Ciao @SirAionTech, mi sa che nel tuo test runboat non avevi entrambe le company abilitate, da questo video puoi vedere come al momento si può selezionare tipi DdT di entrambe le company nel picking di C1: https://recordit.co/wt3QqmnNem

Reply:
Ah non sapevo ci fossero da abilitare più aziende, si possono creare gli step per riprodurre il problema e magari una issue di tracciamento? Vedi https://github.com/OCA/l10n-italy/wiki/Team-di-sviluppo#apertura-issue.

Ho riprovato attivando entrambe le aziende: 

https://github.com/OCA/l10n-italy/assets/143788994/b0d4d3df-6a7f-441e-a48f-f4a2270b1e5c

e viene sempre correttamente mostrata solo l'azienda del DdT che si sta creando.

Nel video che hai mostrato però si sta usando il wizard di creazione del DdT che non viene modificato in questo punto (è l'altra modifica della PR).

Reply:
@SirAionTech hai ragione, eccola https://github.com/OCA/l10n-italy/issues/3730

A livello funzionale il test di tutte queste PR che sistemano il multicompany sono da fare con la doppia azienda attiva (è quello che può indurre in errore l'utente)

Reply:
Ciao, hai pienamente ragione, infatti il problema non era sul delivery note in sè, ma piuttosto sul `stock.picking` che ha il field related ad esso.

Il domain che viene computato attraverso il `check_company=True` non veniva portato avanti attraverso il related, quindi ho messo il `check_company` anche sul field del picking.

Grazie mille per avermi fatto notare l'errore! 

--- 
Per dare maggiore contesto, mi sto riferendo a questo field:
![immagine](https://github.com/OCA/l10n-italy/assets/42804718/4d0f6518-b428-4d85-bcd6-43e94dbde94b)